src/pages/modules/components/ModuleDialog.tsx

```diff
@@ -45,54 +39,108 @@ export function ModuleDialog() {
   const [nameError, setNameError] = React.useState<boolean>(false);
 
-  const [dialogInput, setDialogInput] =
-    React.useState<IModuleBase>(EmptyModule);
+  const [dialogInput, setDialogInput] = React.useState<IModuleBase>(NullModule);
 
   React.useEffect(() => {
     if (action === DialogStatus.EDIT) {
       setDialogInput(module);
     } else {
-      setDialogInput(EmptyModule);
+      setDialogInput(NullModule);
     }
   }, [action, module]);
 
   const handleDialogSubmit = () => {
-    if (
-      numError ||
-      nameError ||
-      isBlank(dialogInput.name) ||
-      isNaN(dialogInput.number)
-    ) {
+    const { name, number } = dialogInput;
+
+    setNameError(isBlank(name));
+    setNumError(isInvalidNum(number));
+
+    if (isBlank(name) || isInvalidNum(number)) {
       return;
-    }
-    if (action === DialogStatus.CREATE) {
+    } else if (action === DialogStatus.CREATE) {
       dispatch(requestNewModule(dialogInput));
     } else if (action === DialogStatus.EDIT) {
       dispatch(requestModifyModule(dialogInput));
     }
   };
 
+  const handleDialogClose = () => {
+    setNumError(false);
+    setNameError(false);
+    dispatch(closeDialog());
+  };
+
   const FooterButtons = [
     {
       label: "Cancel",
-      onClick: () => dispatch(closeDialog()),
+      onClick: handleDialogClose,
       variant: "contained",
       color: "error",
     },
     {
       label: action === DialogStatus.CREATE ? "Add Module" : "Edit Module",
       onClick: () => handleDialogSubmit(),
       variant: "contained",
+      disabled: numError || nameError,
     },
   ];
 
+  const DialogTitle = (status: DialogStatus) => {
+    if (status === DialogStatus.CREATE) {
+      const { name, number } = dialogInput;
+
+      let display = {
+        number: !isInvalidNum(number) ? dialogInput.number : "",
+        name: !isBlank(name) ? dialogInput.name : "",
+      };
+
+      let newModuleTitle: string;
+
+      if (display.name && display.number) {
+        newModuleTitle = `Module ${display.number} - ${display.name}`;
+      } else {
+        newModuleTitle = "New Module";
+      }
+
+      return (
+        <>
+          <Typography variant="button" fontSize="subtitle2" color="primary">
+            Creating a new module
+          </Typography>
+
+          <Typography variant="h5" component="div" fontWeight="bold">
+            {newModuleTitle}
+          </Typography>
+        </>
+      );
+    } else if (status === DialogStatus.EDIT) {
+      return (
+        <>
+          <Typography variant="button" fontSize="subtitle2" color="primary">
+            Editing an existing module
+          </Typography>
+
+          <Typography variant="h5" component="div" fontWeight="bold">
+            {`Module ${module.number} - ${module.name}`}
+          </Typography>
+        </>
+      );
+    } else {
+      return (
+        <Typography variant="h5" component="div" fontWeight="bold">
+          Closing this dialog
+        </Typography>
```

Comment:
I will probably create a generic component for both this dialog and the accounts page to create title components more concisely. 

But I need to spend time seeing how to make it generic and still have it work for every dialog so I will be doing that next PR and pushing current build for now.